src/Swift.Bindings/src/Model/TypeNames/CSharpTypeName.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Represents a C# type name.
+/// </summary>
+public record CSharpTypeName
+{
+    /// <summary>
+    /// The namespace name.
+    /// </summary>
+    public string Namespace { get; }
+
+    /// <summary>
+    /// The type name.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// The fully qualified type name, including namespace.
+    /// </summary>
+    public string FullyQualifiedName { get; }
```

Comment:
I assume this is not assembly qualified, at least make it clear.
It also means that we can't ever rely on this being unique across C# types.